### ISSUE TITLE
Example community with an object-relational mapper

### ISSUE THREAD

**Issue by NielsZeilemaker** [Tribler/dispersy#61]
Outcome: a tutorial with one or all three ORM packages used
Goal: use higher-level relational algebra, instead of hand-coded SQL
Motivation: less code, easy to expand and re-use

Options Sqlalchemy, Storm, Django:
- http://www.sqlalchemy.org/
- https://storm.canonical.com/
- https://www.djangoproject.com/

One of many discussions on this: http://stackoverflow.com/questions/53428/what-are-some-good-python-orm-solutions

FYI: originally created by Johan in the Tribler project Tribler/tribler#130

**Comment by boudewijn-tribler** [comment]
Last time I talked with Johan about this it seemed like we didn't want to put time into ORM anymore.  If this is still the case I suggest we close this issue.  Johan?

**Comment by NielsZeilemaker** [comment]
Let's see if we can get an orm to work in Dispersy. Looking at this sqlobject, it seems pretty trivial:
http://code.saghul.net/index.php/tag/sqlobject/

**Comment by LipuFei** [comment]
@NielsZeilemaker I have a branch on Tribler that changes some database to use an ORM called [Axiom](https://divmod.readthedocs.org/en/latest/products/axiom/axiom-why-axiom.html). I converted all tables expect the Channel-related ones to use Axiom and it worked. So this would also be an option.

**Comment by NielsZeilemaker** [comment]
Could you give me a pointer to this branch?

**Comment by LipuFei** [comment]
https://github.com/LipuFei/tribler/tree/refactory_axiom_db

**Comment by NielsZeilemaker** [comment]
Axiom seems to be something else, its an object store. I'd rather use an ORM if possible, and SQLObject seems a bit more mature.

**Comment by whirm** [comment]
@NielsZeilemaker it can be used as an ORM just fine: https://github.com/twisted/axiom/blob/master/README.txt 
Also, sqlobject doesn't seem to have a working schema upgrade system...

**Comment by NielsZeilemaker** [comment]
If we're not using it as an object database, but as orm then ok. 
But let's start with Dispersy, it's smaller.

**Comment by whirm** [comment]
Yeah, that's what we where talking about with @LipuFei right now.
